src/colorizer/AnnotationData.ts

```diff
@@ -230,4 +275,38 @@ export class AnnotationData implements IAnnotationDataGetters, IAnnotationDataSe
     }
     this.timeToLabelIdMap = null;
   }
+
+  toCsv(dataset: Dataset, delimiter: string = ","): string {
+    const idsToLabels = this.getIdsToLabels();
+    const headerRow = [CSV_COL_ID, CSV_COL_TRACK, CSV_COL_TIME];
+
+    headerRow.push(...this.labelData.map((label) => label.name.trim()));
+
+    const csvRows: number[][] = [];
+    for (const [id, labels] of idsToLabels) {
+      const track = dataset.getTrackId(id);
+      const time = dataset.getTime(id);
+
+      const row = [id, track, time];
+      for (let i = 0; i < this.labelData.length; i++) {
+        row.push(labels.includes(i) ? 1 : 0);
+      }
+      csvRows.push(row);
+    }
+
+    const csvString = Papa.unparse(
+      { fields: headerRow, data: csvRows },
+      { delimiter: delimiter, header: true, escapeFormulae: true }
+    );
+
+    // TODO: Prepend additional metadata (e.g. label colors) when we add support
+    // for importing annotation data?
+
+    // // Add comment metadata for label name + colors
+    // const metadataRows = this.labelData.map((label) => {
+    //   return `# ${label.name},#${label.color.getHexString()}`;
+    // });
```

Comment:
is this commented-out code just extra context for the TODO above it? or can it be deleted?

Reply:
It's prototyping code I had for the above TODO. It's simple though so I'll remove it!